libraries/AP_AHRS/AP_AHRS.cpp

```diff
@@ -1871,13 +1871,15 @@ bool AP_AHRS::get_relative_position_D_origin(float &posD) const
 /*
   return relative position from home in meters
 */
-void AP_AHRS::get_relative_position_D_home(float &posD) const
+bool AP_AHRS::get_relative_position_D_home(float &posD) const
 {
     if (!_home_is_set) {
         // fall back to an altitude derived from barometric pressure
         // differences vs a calibrated ground pressure:
         posD = -AP::baro().get_altitude();
-        return;
+        // note that the Baro library currently lies about its success
+        // in providing an altitude.  Not all vehicles have barometers.
+        return true;
```

Comment:
all returns are true, so bool pointless (in fact misleading)

Reply:
I've removed this change entirely - we continue to `void` now.